android/lib/src/main/java/com/swmansion/gesturehandler/GestureHandler.kt

```diff
@@ -0,0 +1,482 @@
+    if (state == newState) {
+      return
+    }
+    val oldState = state
+    state = newState
+    if (state == STATE_ACTIVE) {
+      // Generate a unique coalescing-key each time the gesture-handler becomes active. All events will have
+      // the same coalescing-key allowing EventDispatcher to coalesce RNGestureHandlerEvents when events are
+      // generated faster than they can be treated by JS thread
+      eventCoalescingKey = sNextEventCoalescingKey++
+    }
+    mOrchestrator!!.onHandlerStateChange(this, newState, oldState)
+    onStateChange(newState, oldState)
+  }
+
+  fun wantEvents(): Boolean {
+    return isEnabled
+      && state != STATE_FAILED
+      && state != STATE_CANCELLED
+      && state != STATE_END
+      && mTrackedPointersCount > 0
+  }
+
+  open fun shouldRequireToWaitForFailure(handler: GestureHandler<*>): Boolean {
+    if (handler === this) {
+      return false
+    }
+
+    return mInteractionController?.shouldRequireHandlerToWaitForFailure(this, handler) ?: false
+  }
+
+  fun shouldWaitForHandlerFailure(handler: GestureHandler<*>): Boolean {
+    if (handler === this) {
+      return false
+    }
+
+    return mInteractionController?.shouldWaitForHandlerFailure(this, handler) ?: false
+  }
+
+  open fun shouldRecognizeSimultaneously(handler: GestureHandler<*>): Boolean {
+    if (handler === this) {
+      return true
+    }
+
+    return mInteractionController?.shouldRecognizeSimultaneously(this, handler) ?: false
+  }
+
+  open fun shouldBeCancelledBy(handler: GestureHandler<*>): Boolean {
+    if (handler === this) {
+      return false
+    }
+
+    return mInteractionController?.shouldHandlerBeCancelledBy(this, handler) ?: false
+  }
+
+  fun isWithinBounds(view: View?, posX: Float, posY: Float): Boolean {
+    var left = 0f
+    var top = 0f
+    var right = view!!.width.toFloat()
+    var bottom = view.height.toFloat()
+    if (mHitSlop != null) {
+      val padLeft = mHitSlop!![HIT_SLOP_LEFT_IDX]
+      val padTop = mHitSlop!![HIT_SLOP_TOP_IDX]
+      val padRight = mHitSlop!![HIT_SLOP_RIGHT_IDX]
+      val padBottom = mHitSlop!![HIT_SLOP_BOTTOM_IDX]
+      if (hitSlopSet(padLeft)) {
+        left -= padLeft
+      }
+      if (hitSlopSet(padTop)) {
+        top -= padTop
+      }
+      if (hitSlopSet(padRight)) {
+        right += padRight
+      }
+      if (hitSlopSet(padBottom)) {
+        bottom += padBottom
+      }
+      val width = mHitSlop!![HIT_SLOP_WIDTH_IDX]
+      val height = mHitSlop!![HIT_SLOP_HEIGHT_IDX]
+      if (hitSlopSet(width)) {
+        if (!hitSlopSet(padLeft)) {
+          left = right - width
+        } else if (!hitSlopSet(padRight)) {
+          right = left + width
+        }
+      }
+      if (hitSlopSet(height)) {
+        if (!hitSlopSet(padTop)) {
+          top = bottom - height
+        } else if (!hitSlopSet(padBottom)) {
+          bottom = top + height
+        }
+      }
+    }
+    return posX in left..right && posY >= top && posY <= bottom
```

Comment:
Did you have a reason for using range operator for one axis and not for the other, or is it just a mistake?

Reply:
This is a check for posY being out of range (inclusive).

Reply:
Resolved in 3ad3f8c.